### PR TITLE
falcon landing demo

### PR DESCRIPTION
**Pull request by vBarbaros**
added some docs files

### CODE REVIEW

File: gym_extensions/discrete/classic/cartpole_contextual.py [hunk 1]
```diff
@@ -63,8 +62,36 @@ def _step(self, action):
...
+
+    def context_space_info(self):
+        context_info_dict = {}
+
+        #context_info_dict['context_vals'] = {
+        #                                'masscart':self.masscart, 
+        #                                'masspole':self.masspole, 
+        #                                'pole_length':self.length, 
+        #                                'force_magnitude':self.force_mag
+        #                                }
```
Comment (Breakend) [inline gym_extensions/discrete/classic/cartpole_contextual.py]: Can you remove all the commented out code like this?

File: gym_extensions/discrete/classic/cartpole_contextual.py~ [hunk 2]
```diff
@@ -0,0 +1,94 @@
+"""
```
Comment (Breakend) [inline gym_extensions/discrete/classic/cartpole_contextual.py~]: can you delete this backup file? anything ending in .py~ should probably go

**Comment by Breakend** [comment]
Also, this seems to have conflicts with the current master branch, can you look into resolving those after you've removed the commented out code and stuff?

**Comment by vBarbaros** [comment]
@Breakend Done, Thanks! (...and let me know if you'd like me to change something/reformulate the way I described the falcon-rocket env in the docs )

File: docs/falcon_rocket.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+## SpaceX-like Falcon Rocket
+
+I've assembled a Falcon rocket that can learn to land itself on the autonomous floating drone ship. I reused a part of the LunarLander (literally, the lander) from <a href="https://github.com/openai/gym">OpenAI Gym</a> and added some more elements so that it makes the task more difficult and more interesting.
```
Comment (Breakend) [inline docs/falcon_rocket.md]: Maybe don't use the first person here, no "I", just describe it and maybe add a reference to your Github or website at the bottom of the page!

Suggested:

This Falcon rocket lander environment encourages landing on an autonomous floating drone ship. It uses part of the LunarLander environment as a base (from <a href="https://github.com/openai/gym">OpenAI Gym</a> ) and adds some elements that make the task more difficult and interesting as follows.

**Comment by Breakend** [comment]
Also, side note, might be interesting to add a fuel cost (like a time penalty?) to the reward. Not sure if this is there already.

**Comment by Breakend** [comment]
Btw, the suggestions about how to change the env aren't really for this pull request, just suggestions. I would say the only change I would suggest before merging in this request is just add a link to your page and maybe some wording in the docs text

**Comment by vBarbaros** [comment]
check lines 355-356, there is the penalty deducted from side_power and main_power consumed; one thing that is not added yet, though, is interrupting the episode `Done=True` if the rocket gets out of the  `x-axis` bounds, will do that asap

File: gym_extensions/continuous/box2d/falcon_drone_ship_lander.py [hunk 4]
```diff
@@ -25,7 +25,7 @@
 MAIN_ENGINE_POWER  = 100.0 # 13.0
 SIDE_ENGINE_POWER  =  3.0 # 0.6
 
-INITIAL_RANDOM = 500 #300.0   # Set 1500 to make game harder
+INITIAL_RANDOM = 500 # Set 1500 to make game harder
 
 FALCON_POLY =[
```
Comment (Breakend) [inline gym_extensions/continuous/box2d/falcon_drone_ship_lander.py]: Btw, i played around with it quickly and something like:


FALCON_POLY =[
    (-4,+150), (-13,135),(-13,0), (-13,-30),
    (+13,-30),(+13,0),(+13,+135), (+4,+150)
    ]

Looks more falcon-y in my opinion (also moving the side engines up to 140.0), but idk if you think this is necessary.

Comment (vBarbaros) [inline gym_extensions/continuous/box2d/falcon_drone_ship_lander.py]: ok, let me try it

Comment (vBarbaros) [inline gym_extensions/continuous/box2d/falcon_drone_ship_lander.py]: I think now you gonna like it (but now I need to change the video :) )

**Comment by Breakend** [comment]
Looks awesome now! Very SpaceX. Could you also update the pic?  at gym-extensions/raw/master/assets/Falcon.png and then i'll merge it in?

**Comment by vBarbaros** [comment]
oh! right

**Comment by vBarbaros** [comment]
done! Thanks for all the feedback, Peter!

**Pull request merged**